bigchaindb/db/backends/rethinkdb.py

```diff
@@ -0,0 +1,379 @@
+
+        Transactions are considered stale if they have been assigned a node,
+        but are still in the backlog after some amount of time specified in the
+        configuration.
+
+        Args:
+            reassign_delay (int): threshold (in seconds) to mark a transaction stale.
+
+        Returns:
+            A cursor of transactions.
+        """
+
+        return self.connection.run(
+                r.table('backlog')
+                .filter(lambda tx: time() - tx['assignment_timestamp'] > reassign_delay))
+
+    def get_transaction_from_block(self, transaction_id, block_id):
+        """Get a transaction from a specific block.
+
+        Args:
+            transaction_id (str): the id of the transaction.
+            block_id (str): the id of the block.
+
+        Returns:
+            The matching transaction.
+        """
+        return self.connection.run(
+                r.table('bigchain', read_mode=self.read_mode)
+                .get(block_id)
+                .get_field('block')
+                .get_field('transactions')
+                .filter(lambda tx: tx['id'] == transaction_id))[0]
+
+    def get_transaction_from_backlog(self, transaction_id):
+        """Get a transaction from backlog.
+
+        Args:
+            transaction_id (str): the id of the transaction.
+
+        Returns:
+            The matching transaction.
+        """
+        return self.connection.run(
+                r.table('backlog')
+                .get(transaction_id)
+                .without('assignee', 'assignment_timestamp')
+                .default(None))
+
+    def get_blocks_status_from_transaction(self, transaction_id):
+        """Retrieve block election information given a secondary index and value
+
+        Args:
+            value: a value to search (e.g. transaction id string, payload hash string)
+            index (str): name of a secondary index, e.g. 'transaction_id'
+
+        Returns:
+            :obj:`list` of :obj:`dict`: A list of blocks with with only election information
+        """
+
+        return self.connection.run(
+                r.table('bigchain', read_mode=self.read_mode)
+                .get_all(transaction_id, index='transaction_id')
+                .pluck('votes', 'id', {'block': ['voters']}))
+
+    def get_transactions_by_metadata_id(self, metadata_id):
+        """Retrieves transactions related to a metadata.
+
+        When creating a transaction one of the optional arguments is the `metadata`. The metadata is a generic
+        dict that contains extra information that can be appended to the transaction.
+
+        To make it easy to query the bigchain for that particular metadata we create a UUID for the metadata and
+        store it with the transaction.
+
+        Args:
+            metadata_id (str): the id for this particular metadata.
+
+        Returns:
+            A list of transactions containing that metadata. If no transaction exists with that metadata it
```

Comment:
If we consider a cursor an `iterable` we are quite safe. But yes, this might be subject to change, I'll wait for some more inputs for MongoDB to decide what's the best approach.